frontend/src/vault/keysign/utils/getPreSignedImageHashes.ts

```diff
@@ -0,0 +1,50 @@
+import { WalletCore } from '@trustwallet/wallet-core';
+
+import { getErc20ApprovePreSignedImageHash } from '../../../chain/evm/tx/getErc20ApprovePreSignedImageHash';
+import { incrementKeysignPayloadNonce } from '../../../chain/evm/tx/incrementKeysignPayloadNonce';
+import { getThorchainSwapPreSignedImageHashes } from '../../../chain/thor/swap/tx/getThorchainSwapPreSignedImageHashes';
+import { KeysignPayload } from '../../../gen/vultisig/keysign/v1/keysign_message_pb';
+import { shouldBePresent } from '../../../lib/utils/assert/shouldBePresent';
+import { Chain } from '../../../model/chain';
+import { BlockchainServiceFactory } from '../../../services/Blockchain/BlockchainServiceFactory';
+
+type Input = {
+  keysignPayload: KeysignPayload;
+  walletCore: WalletCore;
+};
+
+export const getPreSignedImageHashes = async ({
+  keysignPayload,
+  walletCore,
+}: Input): Promise<string[]> => {
+  const coin = shouldBePresent(keysignPayload.coin);
+  const chain = coin.chain as Chain;
+
+  const { erc20ApprovePayload, ...restOfKeysignPayload } = keysignPayload;
+  if (erc20ApprovePayload) {
+    const approveImageHash = getErc20ApprovePreSignedImageHash({
+      keysignPayload,
+      walletCore,
+    });
+
+    const restOfImageHashes = await getPreSignedImageHashes({
+      keysignPayload: incrementKeysignPayloadNonce(
+        new KeysignPayload(restOfKeysignPayload)
+      ),
+      walletCore,
+    });
+
+    return [approveImageHash, ...restOfImageHashes];
```

Comment:
Question: since most transactions will not have approval , thus `approveImageHash` shouldn't exist , will this result in an empty record for other normal send?

Reply:
If the `erc20ApprovePayload` field is not present, the function will simply continue using the blockchain service to generate hashes, and there will not be an approve image hash in the result array.